### PR TITLE
[StatefulTransform] Extract aie.flow generation

### PR DESCRIPTION
**Pull request by jtuyls**
Extracts `aie.flow` logic from `AMDAIEStatefulTransform` to start adding packet flow functionality, see https://github.com/nod-ai/iree-amd-aie/issues/714.

Note: this adds some temporary additions to the AIE dialect (symbol name) that will be removed again in the near future.

### CODE REVIEW

File: compiler/plugins/target/AMD-AIE/aie/AMDAIEObjectFifoStatefulTransform.cpp [hunk 1]
```diff
@@ -569,6 +552,7 @@ void splitFifo(
   assert(!splitConsumerFifos.empty() &&
          "expected some fifos to have been split");
   splitFifos.emplace_back(createOp, splitConsumerFifos);
+  // splitFifoOrigs.push_back();
```
Comment (makslevental) [inline compiler/plugins/target/AMD-AIE/aie/AMDAIEObjectFifoStatefulTransform.cpp:555]: Remove

Comment (jtuyls) [reply compiler/plugins/target/AMD-AIE/aie/AMDAIEObjectFifoStatefulTransform.cpp]: Thanks, done

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/assign_channels.mlir [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+// CHECK:         %[[CHANNEL_1:.+]] = amdaie.channel(%[[tile_0_1]], 0)
+// CHECK:         amdaie.connection(%{{.+}} {%[[CHANNEL_1]]}, %{{.+}} {%[[CHANNEL_0]]})
+// CHECK:         %[[CHANNEL_2:.+]] = amdaie.channel(%[[tile_0_0]], 1)
+// CHECK:         %[[CHANNEL_3:.+]] = amdaie.channel(%[[tile_0_1]], 1)
+// CHECK:         amdaie.connection(%{{.+}} {%[[CHANNEL_3]]}, %{{.+}} {%[[CHANNEL_2]]})
+// CHECK:         %[[CHANNEL_4:.+]] = amdaie.channel(%[[tile_0_0]], 2)
+// CHECK:         %[[CHANNEL_5:.+]] = amdaie.channel(%[[tile_0_1]], 2)
+// CHECK:         amdaie.connection(%{{.+}} {%[[CHANNEL_5]]}, %{{.+}} {%[[CHANNEL_4]]})
+#executable_target_amdaie_xclbin_fb = #hal.executable.target<"amd-aie", "amdaie-xclbin-fb", {target_device = "npu1_4col", ukernels = "none"}>
+module attributes {hal.executable.target = #executable_target_amdaie_xclbin_fb} {
```
Comment (Abhishek-Varma) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/assign_channels.mlir:19]: You can remove these from the lit tests as this pass is not basing any operations on `ModuleOp`'s attributes.

**Review by Abhishek-Varma — APPROVED** [review]
Really good! Thanks!

File: compiler/plugins/target/AMD-AIE/aie/AMDAIEObjectFifoStatefulTransform.cpp [hunk 3]
```diff
@@ -800,78 +783,92 @@ void createBuffersAndLocks(
...
 
+  // Collect producer and consumer DMA channels
+  if (!symbolToFlowOps.contains(producer.getSymName())) {
+    return producer.emitOpError()
+           << "symbol name not found in symbol to flow ops map";
+  }
+  SmallVector<FlowOp> flowOps = symbolToFlowOps.at(producer.getSymName());
+  SmallVector<uint8_t> producerChannelsVec = llvm::map_to_vector(
+      flowOps, [](FlowOp flowOp) { return flowOp.getSourceChannel(); });
+  llvm::SmallSetVector<uint8_t, 1> producerChannels(producerChannelsVec.begin(),
```
[the remark this answers is not in the document]
Comment (jtuyls) [reply compiler/plugins/target/AMD-AIE/aie/AMDAIEObjectFifoStatefulTransform.cpp:820]: With `SmallSetVector` you can't just drop the `llvm::` namespace as it's not included in the mlir namespace by default like other data structures (`SmallVector` etc). I could include it in this file, but I am keeping `llvm::` for now for consistency as that's how we have used it in other places within iree-amd-aie and it's commonly used that way in iree as well.

**Pull request merged**